pkg/core/blockchain.go

```diff
@@ -1262,37 +1198,45 @@ func (bc *Blockchain) GetValidators(txes ...*transaction.Transaction) ([]*keys.P
 					if prevOutput.AssetID.Equals(governingTokenTX().Hash()) {
 						if len(accountState.Votes) > 0 {
 							for _, vote := range accountState.Votes {
-								validatorState, err := chainState.validators.getAndUpdate(chainState.store, vote)
+								validatorState, err := bc.dao.GetValidatorStateOrNew(vote)
 								if err != nil {
 									return nil, err
 								}
 								validatorState.Votes -= prevOutput.Amount
+								if err = bc.dao.PutValidatorState(validatorState); err != nil {
+									return nil, err
+								}
```

Comment:
Is it the third time I'm seeing it?

Reply:
Still relevant, this branch is for `else` down below and probably this logic should be refactored away into a separate function, because it really is a third duplicate.